thrall/app/lib/elasticsearch/ElasticSearch.scala

```diff
@@ -295,19 +339,17 @@ class ElasticSearch(
       nestedQuery("usages").query(existsQuery("usages"))
     )
 
-    val eventualDeleteResponse = executeAndLog(count(imagesCurrentAlias).query(deletableImage), s"ES6 searching for image to delete: $id").flatMap { r =>
-      val deleteFuture = r.result.count match {
-        case 1 => executeAndLog(deleteById(imagesCurrentAlias, id), s"ES6 deleting image $id")
-        case _ => Future.failed(ImageNotDeletable)
-      }
-      deleteFuture
-        .incrementOnSuccess(metrics.map(_.deletedImages))
-        .incrementOnFailure(metrics.map(_.failedDeletedImages)) { case ImageNotDeletable => true }
+    // Uses the cached migration status to determine if a migration is in progress: if there is not, this will be None
+    // and get flattened out
+    val migrationAliasIfActive = migrationStatus match {
+      case InProgress(_) => Some(imagesMigrationAlias)
+      case _ => None
     }
-
-    List(eventualDeleteResponse.map { _ =>
-      ElasticSearchDeleteResponse()
-    })
+    List(Some(imagesCurrentAlias), migrationAliasIfActive)
+      .flatten
+      .map { indexAlias => deleteFromIndex(id, indexAlias, deletableImage).map { _ =>
+        ElasticSearchDeleteResponse()
+      }}
```

Comment:
why do we need to use the migration alias here, when we have the actual index name? does `deleteFromIndex` not like real index names??

Reply:
I don't think there was any specific reason to use the alias over the index name, they should be equivalent AFAIK. I can switch it to the name if there's a strong preference for that

Reply:
not a strong preference, just we're ignoring the value with an `_` when we could perhaps simplify the whole block and eliminate the variable, something like... 
```scala
migrationStatus match {
  case InProgress(migrationIndexName) => List(imagesCurrentAlias, migrationIndexName)
  case _ => List(imagesCurrentAlias)
}.map { index => 
  deleteFromIndex(id, index, deletableImage).map { _ => ElasticSearchDeleteResponse() }
}
```
(this avoids the implicit conversion too)